src/java.base/share/classes/jdk/internal/classfile/impl/AbstractPoolEntry.java

```diff
@@ -0,0 +1,1173 @@
+                            }
+                            char v = (char) (((c & 0x0F) << 12) | ((char2 & 0x3F) << 6) | ((char3 & 0x3F) << 0));
+                            chararr[chararr_count++] = v;
+                            hash = 31 * hash + v;
+                            break;
+                        }
+                        default:
+                            // 10xx xxxx,  1111 xxxx
+                            throw new CpException("malformed input around byte " + px);
+                    }
+                }
+                this.hash = hashString(hash);
+                charLen = chararr_count;
+                this.chars = chararr;
+                state = State.CHAR;
+            }
+
+        }
+
+        @Override
+        public Utf8EntryImpl clone(ConstantPoolBuilder cp) {
+            if (cp.canWriteDirect(constantPool))
+                return this;
+            return (state == State.STRING && rawBytes == null)
+                   ? (Utf8EntryImpl) cp.utf8Entry(stringValue)
+                   : ((SplitConstantPool) cp).maybeCloneUtf8Entry(this);
+        }
+
+        @Override
+        public int hashCode() {
+            if (state == State.RAW)
+                inflate();
+            return hash;
+        }
+
+        @Override
+        public String toString() {
+            if (state == State.RAW)
+                inflate();
+            if (state != State.STRING) {
+                stringValue = (chars != null)
+                              ? new String(chars, 0, charLen)
+                              : new String(rawBytes, offset, charLen, StandardCharsets.UTF_8);
+                state = State.STRING;
+            }
+            return stringValue;
+        }
+
+        @Override
+        public String stringValue() {
+            return toString();
+        }
+
+        @Override
+        public ConstantDesc constantValue() {
+            return stringValue();
+        }
+
+        @Override
+        public int length() {
+            if (state == State.RAW)
+                inflate();
+            return charLen;
+        }
+
+        @Override
+        public char charAt(int index) {
+            if (state == State.STRING)
+                return stringValue.charAt(index);
+            if (state == State.RAW)
+                inflate();
+            return (chars != null)
+                   ? chars[index]
+                   : (char) rawBytes[index + offset];
+        }
+
+        @Override
+        public CharSequence subSequence(int start, int end) {
+            return toString().subSequence(start, end);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) return true;
+            if (o instanceof Utf8EntryImpl u) {
+                return equalsUtf8(u);
+            } else if (o instanceof Utf8Entry u) {
+                return equalsString(u.stringValue());
+            }
```

Comment:
Dead branch? since there is only one concrete implementation of `Utf8Entry`.

Reply:
fixed, thanks.